rpc-client-impl_test.go

```diff
@@ -73,42 +73,93 @@ func (rpcClient *ReconnectRPCClient) Call(serviceMethod string, args interface{}
 	// rpc.Client for a subsequent reconnect.
 	err := rpcClient.rpc.Call(serviceMethod, args, reply)
 	if IsRPCError(err) {
+		rpcClient.rpc.Close()
 		rpcClient.rpc = nil
 	}
 	return err
 }
 
 func (rpcClient *ReconnectRPCClient) RLock(args LockArgs) (status bool, err error) {
 	err = rpcClient.Call("Dsync.RLock", &args, &status)
+	if err == rpc.ErrShutdown {
+		if ierr := rpcClient.Init(); ierr != nil {
+			return false, ierr
+		}
+		err = rpcClient.Call("Dsync.RLock", &args, &status)
+	}
 	return status, err
 }
 
 func (rpcClient *ReconnectRPCClient) Lock(args LockArgs) (status bool, err error) {
 	err = rpcClient.Call("Dsync.Lock", &args, &status)
+	if err == rpc.ErrShutdown {
+		if ierr := rpcClient.Init(); ierr != nil {
+			return false, ierr
+		}
+		err = rpcClient.Call("Dsync.Lock", &args, &status)
+	}
 	return status, err
 }
 
 func (rpcClient *ReconnectRPCClient) RUnlock(args LockArgs) (status bool, err error) {
 	err = rpcClient.Call("Dsync.RUnlock", &args, &status)
+	if err == rpc.ErrShutdown {
+		if ierr := rpcClient.Init(); ierr != nil {
+			return false, ierr
+		}
+		err = rpcClient.Call("Dsync.RUnlock", &args, &status)
+	}
 	return status, err
 }
 
 func (rpcClient *ReconnectRPCClient) Unlock(args LockArgs) (status bool, err error) {
 	err = rpcClient.Call("Dsync.Unlock", &args, &status)
+	if err == rpc.ErrShutdown {
```

Comment:
Can't this be done as part of rpcClient's Call method? This is being repeated for every other method that depends on Call.